### DISCUSSION TITLE
Describe GDExtension interface in a machine readable format

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#9524] [Scripting]
Instead of using `gdextension_interface.h` the C header file, describe GDExtension interface in a machine readable format like JSON, XML, etc.

This makes it easier for other languages to generate wrappers containing routine code automatically.

Vulkan interface example: [vk.xml](https://raw.githubusercontent.com/KhronosGroup/Vulkan-Headers/main/registry/vk.xml).

**Comment by dsnopek** [comment]
Yeah, this could make sense. It's done by a number of other standard API, so we'd be in good company. We already have a number of restrictions/requirements around what can be put in `gdextension_interface.h`, so having it be generated from XML (or some other format) would help to formalize that. So, I'm personally very open to this idea! Like all things in Open Source, we'd need someone to take on doing that work ;-)
